src/components/Layouts/Primary/TopBar/Items/AddMenu/Items/CreateContact/CreateContact.tsx

```diff
@@ -0,0 +1,208 @@
+import { useCreateContactMutation } from './CreateContact.generated';
+
+interface Props {
+  accountListId: string;
+  handleClose: () => void;
+}
+
+const CreateContactTitle = styled(DialogTitle)(() => ({
+  textTransform: 'uppercase',
+  textAlign: 'center',
+}));
+
+const CloseButton = styled(IconButton)(({ theme }) => ({
+  position: 'absolute',
+  right: theme.spacing(1),
+  top: theme.spacing(1),
+  color: theme.palette.text.primary,
+  '&:hover': {
+    backgroundColor: '#EBECEC',
+  },
+}));
+
+const LogFormControl = styled(FormControl)(() => ({
+  width: '100%',
+}));
+
+const LogFormLabel = styled(FormLabel)(({ theme }) => ({
+  margin: theme.spacing(1, 0),
+  fontWeight: 'bold',
+  color: theme.palette.primary.dark,
+  '& span': {
+    color: theme.palette.error.main,
+  },
+}));
+
+const LogTextField = styled(TextField)(({ theme }) => ({
+  '& div': {
+    padding: theme.spacing(1),
+  },
+}));
+
+const LoadingIndicator = styled(CircularProgress)(({ theme }) => ({
+  margin: theme.spacing(0, 1, 0, 0),
+}));
+
+const contactSchema: yup.SchemaOf<ContactCreateInput> = yup.object({
+  churchName: yup.string().nullable(),
+  envelopeGreeting: yup.string().nullable(),
+  greeting: yup.string().nullable(),
+  id: yup.string().nullable(),
+  likelyToGive: yup.mixed<LikelyToGiveEnum>().nullable(),
+  locale: yup.string().nullable(),
+  name: yup.string().required(),
+  nextAsk: yup.string().nullable(),
+  noAppeals: yup.boolean().nullable(),
+  notes: yup.string().nullable(),
+  pledgeAmount: yup.number().nullable(),
+  pledgeCurrency: yup.string().nullable(),
+  pledgeFrequency: yup.mixed<PledgeFrequencyEnum>().nullable(),
+  pledgeReceived: yup.boolean().nullable(),
+  pledgeStartDate: yup.string().nullable(),
+  preferredContactMethod: yup.mixed<PreferredContactMethodEnum>().nullable(),
+  sendNewsletter: yup.mixed<SendNewsletterEnum>().nullable(),
+  starred: yup.boolean().nullable(),
+  status: yup.mixed<StatusEnum>().nullable(),
+  tagList: yup.array().of(yup.string()).default([]),
+  timezone: yup.string().nullable(),
+  userId: yup.string().nullable(),
+  website: yup.string().nullable(),
+});
+
+const CreateContact = ({
+  accountListId,
+  handleClose,
+}: Props): ReactElement<Props> => {
+  const { t } = useTranslation();
+  const { enqueueSnackbar } = useSnackbar();
+  const { push } = useRouter();
+
+  const initialContact: ContactCreateInput = {
+    name: '',
+  };
+
+  const [createContact, { loading: creating }] = useCreateContactMutation();
+
+  const onSubmit = async (attributes: ContactCreateInput) => {
+    try {
+      await createContact({
+        variables: {
+          accountListId,
+          attributes,
+        },
+        update: (_cache, { data }) => {
+          const contactId = data?.createContact?.contact.id;
+          if (contactId) {
+            push({
+              pathname: '/accountLists/[accountListId]/contacts/[contactId]',
+              query: { accountListId, contactId },
+            });
```

Comment:
The current logic in the angular app is to navigate that user to the newly created contact upon successful creation. Implemented that here as well.